samples/nrf9160/modem_shell/src/startup_cmd/startup_cmd_settings.c

```diff
@@ -0,0 +1,257 @@
+		err = read_cb(cb_arg, &startup_cmd_data.cmd1_str,
+			      sizeof(startup_cmd_data.cmd1_str));
+		if (err < 0) {
+			mosh_error("Failed to read set_cmd #1, error: %d", err);
+			return err;
+		}
+	} else if (strcmp(key, STARTUP_CMD_2_KEY) == 0) {
+		err = read_cb(cb_arg, &startup_cmd_data.cmd2_str,
+			      sizeof(startup_cmd_data.cmd2_str));
+		if (err < 0) {
+			mosh_error("Failed to read set_cmd #2, error: %d", err);
+			return err;
+		}
+	} else if (strcmp(key, STARTUP_CMD_3_KEY) == 0) {
+		err = read_cb(cb_arg, &startup_cmd_data.cmd3_str,
+			      sizeof(startup_cmd_data.cmd3_str));
+		if (err < 0) {
+			mosh_error("Failed to read set_cmd #3, error: %d", err);
+			return err;
+		}
+	}
+
+	return 0;
+}
+
+/* ****************************************************************************/
+
+char *startup_cmd_get(uint8_t mem_slot)
+{
+	if (mem_slot == 1) {
+		return startup_cmd_data.cmd1_str;
+	} else if (mem_slot == 2) {
+		return startup_cmd_data.cmd2_str;
+	} else if (mem_slot == 3) {
+		return startup_cmd_data.cmd3_str;
+	}
+
+	mosh_error("%s:unsupported memory slot %d", __func__, mem_slot);
+	return NULL;
+}
+
+int startup_cmd_save(const char *cmd_str, int mem_slot)
+{
+	int err = 0;
+	const char *key;
+	int len = strlen(cmd_str);
+	char *tmp_ptr = NULL;
+
+	assert(mem_slot >= 1 && mem_slot <= STARTUP_CMD_MAX_COUNT);
+
+	if (len > STARTUP_CMD_MAX_LEN) {
+		mosh_error("%s: Too long cmd, max is %d", (__func__), STARTUP_CMD_MAX_LEN);
+		return -EINVAL;
+	}
+
+	if (mem_slot == 1) {
+		key = STARTUP_CMD_SETTINGS_KEY "/" STARTUP_CMD_1_KEY;
+		tmp_ptr = startup_cmd_data.cmd1_str;
+	} else if (mem_slot == 2) {
+		key = STARTUP_CMD_SETTINGS_KEY "/" STARTUP_CMD_2_KEY;
+		tmp_ptr = startup_cmd_data.cmd2_str;
+	} else if (mem_slot == 3) {
+		key = STARTUP_CMD_SETTINGS_KEY "/" STARTUP_CMD_3_KEY;
+		tmp_ptr = startup_cmd_data.cmd3_str;
+	} else {
+		mosh_error("%s: unsupported memory slot %d", (__func__), mem_slot);
+		err = -EINVAL;
+		return err;
+	}
+	err = settings_save_one(key, cmd_str, len + 1);
+	if (err) {
+		mosh_error("%s: cannot save command %s to settings."
+			   "err %d from settings_save_one()",
+			   cmd_str, (__func__), err);
+		return err;
+	}
+	strcpy(tmp_ptr, cmd_str);
+	mosh_print("%s: key %s with value %s saved", (__func__), key, cmd_str);
+
+	return 0;
+}
+
+int startup_cmd_clear(uint8_t mem_slot)
+{
+	int err;
+	const char *key;
+	char *tmp_ptr = NULL;
+
+	if (mem_slot == 1) {
+		key = STARTUP_CMD_SETTINGS_KEY "/" STARTUP_CMD_1_KEY;
+		tmp_ptr = startup_cmd_data.cmd1_str;
+	} else if (mem_slot == 2) {
+		key = STARTUP_CMD_SETTINGS_KEY "/" STARTUP_CMD_2_KEY;
+		tmp_ptr = startup_cmd_data.cmd2_str;
+	} else if (mem_slot == 3) {
+		key = STARTUP_CMD_SETTINGS_KEY "/" STARTUP_CMD_3_KEY;
+		tmp_ptr = startup_cmd_data.cmd3_str;
+	} else {
+		mosh_error(
+			"link_sett_delete_normal_mode_at_cmd_str: unsupported memory slot %d",
+			mem_slot);
+		return -EINVAL;
+	}
+
+	err = settings_save_one(key, '\0', 1);
```

Comment:
This clears the command from settings, but looks like it's not removed from RAM?

Reply:
Actually, it seems that this startup_cmd_clear() is not even used anymore. Clearing is handled in startup_cmd_save().
Goot catch!